build-tools/automation/build.linux.groovy

```diff
@@ -0,0 +1,220 @@
+                dir (XADir) {
+                    if (retryAttempt > 0) {
+                        echo "WARNING : Stage checkout failed on try #${retryAttempt}. Waiting ${waitSecondsBeforeRetry} seconds"
+                        sleep(waitSecondsBeforeRetry)
+                        echo "Retrying ..."
+                        waitSecondsBeforeRetry = waitSecondsBeforeRetry * 2
+                    }
+
+                    retryAttempt++
+                    scmVars = checkout scm
+                }
+            }
+        }
+
+        def workspace = "/mnt/jenkins/workspace/${env.JOB_BASE_NAME}"
+        utils = load "${workspace}/${XADir}/build-tools/automation/utils.groovy"
+
+        utils.stageWithTimeout('init', 30, 'MINUTES', XADir, true) {    // Typically takes less than a second for CI builds; execChRootCommand execution can take several minutes for PR builds
+            // Note: PR plugin environment variable settings available here: https://wiki.jenkins.io/display/JENKINS/GitHub+pull+request+builder+plugin
+            isPr = env.ghprbActualCommit != null
+            isStable = env.IsStable == '1'
+            def branch = isPr ? env.GIT_BRANCH : scmVars.GIT_BRANCH
+            def commit = isPr ? env.ghprbActualCommit : scmVars.GIT_COMMIT
+
+            def buildType = isPr ? 'PR' : 'CI'
+
+            echo "HostName: ${env.NODE_NAME}"
+            echo "Git repo: ${env.GitRepo}"     // Defined as an environment variable in the jenkins build definition
+            echo "Job: ${env.JOB_BASE_NAME}"
+            echo "Branch: ${branch}"
+            echo "Commit: ${commit}"
+            echo "Build type: ${buildType}"
+            echo "Stable build workflow: ${isStable}"
+
+            pBuilderBindMounts = "/home/${env.USER}"
+            echo "pBuilderBindMounts: ${pBuilderBindMounts}"
+
+            if (isPr) {
+                echo "PR id: ${env.ghprbPullId}"
+                echo "PR link: ${env.ghprbPullLink}"
+
+                // Clear out the PR title and description. This is the equivalent of $JENKINS_HOME/global-pre-script/remove-problematic-ghprb-parameters.groovy used by freestyle builds
+                echo "Clearing the PR title and description environment variables to avoid any special characters contained within from tripping up the build"
+                env.ghprbPullTitle = ''
+                env.ghprbPullLongDescription = ''
+
+                chRootPackages = 'mono-devel autoconf automake build-essential vim-common p7zip-full cmake gettext libtool libgdk-pixbuf2.0-dev intltool pkg-config ruby scons wget xz-utils git nuget ca-certificates-mono g++-mingw-w64 gcc-mingw-w64 libzip4 openjdk-8-jdk unzip lib32stdc++6 lib32z1 libtinfo-dev:i386 linux-libc-dev:i386 zlib1g-dev:i386 gcc-multilib g++-multilib referenceassemblies-pcl zip libz-mingw-w64-dev msbuild mono-csharp-shell libsqlite3-dev sqlite3 cli-common-dev curl'
```

Comment:
why are the packages different here than in the global variable?

Reply:
This is a reflection of the difference for the `chroot builder` configuration differences between the `xamarin-android-linux` and `xamarin-android-linux-pr-builder` workflows.  I simply copied in carbon copies of those settings and so this is a good opportunity to review them.  The PR workflow does not include the following packages relative to `xamarin-android-linux`: `xvfb xauth devscripts fakeroot debhelper libc++-dev`.  Also, the PR workflow uses `libzip4` whereas the CI workflow uses `libzip-dev`. @directhex, @jonp - should the package lists align here?
I see there has been a recent change to `git` package to use `git/stretch-backports` instead and so incorporated that change.  commit https://github.com/xamarin/xamarin-android/pull/2838/commits/53a2f96329a8c5ea4930b90f09e012dd7d14aec3

Reply:
`xvfb` provides a headless framebuffer target, for any tests requiring a display
`xauth` is needed to force output onto an arbitrary display
`devscripts`, `fakeroot` and `debheper` are needed for building Debian packages
I honestly don't know why we added libc++-dev, maybe when we were using a too-old libstdc++?

Reply:
After a chat w/ @directhex, aligning on a single package list for all Linux pipeline workflows (CI and PR). favoring `libzip-dev` over `libzip4`. commit https://github.com/xamarin/xamarin-android/pull/2838/commits/6a4c1a5fb229f0015b739c632ba1f54927f02601